### DISCUSSION TITLE
[Types] Change `SchemaTypeOptions.index` to use `IndexDirection`?

### DISCUSSION THREAD

**Discussion by hasezoey** [Automattic/mongoose#12395] [General]
Basically what the title says, currently `SchemaTypeOptions.index` uses a custom definition of all index options, which could just as easily be used from `IndexDirection` to have it in one place for maintainability, is there any reason why `SchemaTypeOptions.index` uses a custom definition over `IndexDirection`?

**Comment by vkarpov15** [comment]
Nope, no particular reason that I'm aware of. Feel free to give this a shot.

**Reply by hasezoey** [reply]
done with #12482
